### PR TITLE
Add WebIDLs for Filesystem Access API

### PR DESCRIPTION
**Pull request by Speak2Erase**
[IDL](https://fs.spec.whatwg.org/), [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API)

**I'm pretty new to contributing to projects like this, please lmk if I did something wrong!**

I followed the basic instructions listed in `web-sys`'s Readme. 

As far as I can tell, the bindings are correct. However, I don't really know what I am doing, *especially* about `[Throws]`, so that assumption could be wrong.

The Filesystem Access API seems to be pretty stable, however some browsers do not support it (i.e. Firefox) so I added it to unstable.

### CODE REVIEW

**Review by daxpedda — CHANGES_REQUESTED** [review]
LGTM otherwise.
Needs a rebase though.

**Comment by djozis** [comment]
Hopefully not a faux pas, but I wanted this too, including the PR feedback, so I made the changes for my personal use, and thought I'd offer it back here: #3427

**Comment by daxpedda** [comment]
Closing in favor of #3427.

File: crates/web-sys/webidls/unstable/FilesystemAccess.webidl [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  (BufferSource or Blob or USVString)? data;
+};
+
+typedef (BufferSource or Blob or USVString or WriteParams) FileSystemWriteChunkType;
+
+[Exposed=(Window,Worker), SecureContext]
+interface FileSystemWritableFileStream : WritableStream {
+  Promise<undefined> write(FileSystemWriteChunkType data);
+  Promise<undefined> seek(unsigned long long position);
+  Promise<undefined> truncate(unsigned long long size);
```
[the remark this answers is not in the document]
Comment (djozis) [reply crates/web-sys/webidls/unstable/FilesystemAccess.webidl:75]: I created a PR for this, can continue discussion there perhaps: #3435

**Pull request closed without merging**